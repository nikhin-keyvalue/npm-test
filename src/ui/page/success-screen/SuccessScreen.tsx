import Preact, { h } from 'preact';

import Button from 'ui/components/button/Button';
import SuccessTimer from 'ui/components/success-timer/SuccessTimer';
import { ButtonType, PaymentStatus } from 'constants/common';
import { parseAddress } from 'ui/utils/common';
import { ASSETS_PATH } from 'config';

import './style.scss';

type SuccessScreenType = {
  handlePaymentCompleted: (status: string) => void;
  orderId: string;
  account?: string | null;
  successPageDisplayTime?: number;
};

const SuccessScreen: Preact.FunctionComponent<SuccessScreenType> = ({
  handlePaymentCompleted,
  successPageDisplayTime,
  account,
  orderId
}) => (
  <div className='success-screen-container'>
    <img src={`${ASSETS_PATH}/public/icons/pending-icon.svg`} className='success-screen-logo' alt='Pending Icon' />
    <div className='success-screen-status-wrapper'>
      <div className='success-screen-message-wrapper'>
        <div className='success-screen-title'>Payment Processing</div>
        <div className='success-screen-details'>You will be redirected to Merchant site soon</div>
      </div>
      <div className='success-screen-order-wrapper'>
        {account && (
          <div className='success-screen-order-details'>
            <div className='success-screen-order-title'>Account</div>
            <div className='success-screen-order-data-'>{parseAddress(account)}</div>
          </div>
        )}
        <div className='success-screen-order-details'>
          <div className='success-screen-order-title'>Order ID</div>
          <div className='success-screen-order-data'>{orderId}</div>
        </div>
      </div>
    </div>
    <div className='success-screen-button-container'>
      <Button
        buttonLabel='Done'
        type={ButtonType.SECONDARY}
        width={'100px'}
        handleButtonClick={() => handlePaymentCompleted(PaymentStatus.SUCCESS)}
      />
    </div>
    <div className='success-screen-redirection-container'>
      <SuccessTimer
        timeDuration={successPageDisplayTime}
        handlePaymentCompleted={() => handlePaymentCompleted(PaymentStatus.SUCCESS)}
      />
      <div className='success-screen-redirection-label'>Redirecting to Probo</div>
    </div>
  </div>
);

SuccessScreen.defaultProps = {
  successPageDisplayTime: 10
};

export default SuccessScreen;
