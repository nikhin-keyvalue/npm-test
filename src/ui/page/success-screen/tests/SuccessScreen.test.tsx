import { h } from 'preact';
import { configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Adapter } from 'enzyme-adapter-preact-pure';

import SuccessScreen from '../SuccessScreen';

configure({ adapter: new Adapter() });

describe('Counter', () => {
  it('render components', () => {
    const wrapper = mount(<SuccessScreen handlePaymentCompleted={() => {}} orderId={''} account={''} />);
    const tree = shallowToJson(wrapper);

    expect(tree).toMatchSnapshot();
  });
});
