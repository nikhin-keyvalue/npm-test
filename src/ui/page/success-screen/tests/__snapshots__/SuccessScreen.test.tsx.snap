// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter render components 1`] = `
<SuccessScreen
  account=""
  handlePaymentCompleted={[Function]}
  orderId=""
  successPageDisplayTime={10}
>
  <div
    className="success-screen-container"
  >
    <img
      alt="Pending Icon"
      className="success-screen-logo"
      src="undefined/public/icons/pending-icon.svg"
    />
    <div
      className="success-screen-status-wrapper"
    >
      <div
        className="success-screen-message-wrapper"
      >
        <div
          className="success-screen-title"
        >
          Payment Processing
        </div>
        <div
          className="success-screen-details"
        >
          You will be redirected to Merchant site soon
        </div>
      </div>
      <div
        className="success-screen-order-wrapper"
      >
        <div
          className="success-screen-order-details"
        >
          <div
            className="success-screen-order-title"
          >
            Order ID
          </div>
          <div
            className="success-screen-order-data"
          />
        </div>
      </div>
    </div>
    <div
      className="success-screen-button-container"
    >
      <Button
        buttonLabel="Done"
        handleButtonClick={[Function]}
        isDisabled={false}
        type="secondary"
        width="100px"
      >
        <button
          className="zamp-secondary-button"
          disabled={false}
          onClick={[Function]}
          style={
            Object {
              "height": undefined,
              "width": "100px",
            }
          }
        >
          Done
        </button>
      </Button>
    </div>
    <div
      className="success-screen-redirection-container"
    >
      <SuccessTimer
        handlePaymentCompleted={[Function]}
        timeDuration={10}
      >
        <div
          className="base-timer"
        >
          <svg
            className="base-timer__svg"
            viewBox="0 0 100 100"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              className="base-timer__circle"
            >
              <path
                className="base-timer__path-remaining"
                d="
          M 50, 50
          m -45, 0
          a 45,45 0 1,0 90,0
          a 45,45 0 1,0 -90,0
        "
                id="base-timer-path-remaining"
                stroke-dasharray="283"
                strokeDasharray="283"
              />
            </g>
          </svg>
          <span
            className="base-timer__label"
            id="base-timer-label"
          >
            10
          </span>
        </div>
      </SuccessTimer>
      <div
        className="success-screen-redirection-label"
      >
        Redirecting to Probo
      </div>
    </div>
  </div>
</SuccessScreen>
`;
